Simplenote/SPAppDelegate.m

```diff
@@ -618,6 +618,21 @@ - (void)indexSpotlightItems
 
 - (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options
 {
+    if (!self.simperium.user.authenticated) {
+        if ([WPAuthHandler isWPAuthenticationUrl: url]) {
+            SPUser *newUser = [WPAuthHandler authorizeSimplenoteUserFromUrl:url forAppId:[SPCredentials simperiumAppID]];
+            if (newUser != nil) {
+                self.simperium.user = newUser;
+                [self.navigationController dismissViewControllerAnimated:YES completion:nil];
+                [self.simperium authenticationDidSucceedForUsername:newUser.email token:newUser.authToken];
+
+                [SPTracker trackWPCCLoginSucceeded];
+            }
+        }
+
+        return YES;
+    }
```

Comment:
REALLY Nipticky!! Maybe avoid the nested `if(s)` (by moving the first block to another method?)

```
    if (!self.simperium.user.authenticated) {
         [self performDotcomAuthenticationWithURL:url];
         return YES;
     }
```

(or so?)

Reply:
I have extracted it to a separate method (https://github.com/Automattic/simplenote-ios/pull/1088/commits/404f7df92e762a19d269d501d1c2ad64e2b029ef), looks better! Thank you for the suggestion 👍